src/community/features-autopopulate/features-autopopulate-core/src/main/java/org/geoserver/autopopulate/AutopopulateTemplateLoader.java

```diff
@@ -0,0 +1,127 @@
+    /**
+     * Feature type directory to load template against. Its presence is mutually exclusive with
+     * coverageName
+     */
+    protected ResourceInfo resource;
+    /** GeoServer data directory */
+    GeoServerDataDirectory dd;
+    /** Allows for workspace-specific lookup */
+    WorkspaceInfo workspace;
+
+    /**
+     * Constructs the template loader.
+     *
+     * @param rl The geoserver resource loader
+     * @param workspace The workspace to load the template from
+     * @param resource The resource to load the template from
+     */
+    public AutopopulateTemplateLoader(
+            GeoServerResourceLoader rl, WorkspaceInfo workspace, ResourceInfo resource) {
+        this(
+                rl == null
+                        ? new GeoServerDataDirectory(
+                                GeoServerExtensions.bean(GeoServerResourceLoader.class))
+                        : new GeoServerDataDirectory(rl),
+                workspace,
+                resource);
+    }
+
+    /**
+     * Constructs the template loader.
+     *
+     * @param dd The geoserver data directory
+     * @param workspace The workspace to load the template from
+     * @param resource The resource to load the template from
+     */
+    public AutopopulateTemplateLoader(
+            GeoServerDataDirectory dd, WorkspaceInfo workspace, ResourceInfo resource) {
+        this.dd = dd;
+        this.workspace = workspace;
+        this.resource = resource;
+    }
+
+    /**
+     * Load the template from the file system.
+     *
+     * @param path The path to the template
+     * @return The AutopopulateTemplate object
+     * @throws IOException If the template cannot be loaded
+     */
+    public AutopopulateTemplate loadTemplate(String path) throws IOException {
+        File template = null;
+
+        // template look up order
+        // 1. Relative to resource
+        // 2. Relative to store of the resource
+        // 3. Relative to workspace of resource
+        // 4. Relative to workspaces directory
+        if (resource != null) {
+            // first check relative to set resource
+            template = Resources.file(dd.get(resource, path));
+
+            if (template == null) {
+                // next try relative to the store
+                template = Resources.file(dd.get(resource.getStore(), path));
+            }
+
+            if (template == null) {
+                // next try relative to the workspace
+                template = Resources.file(dd.get(resource.getStore().getWorkspace(), path));
+            }
+
+            if (template == null) {
+                // try global supplementary files
+                template = Resources.file(dd.getWorkspaces(path));
+            }
+
+            if (template != null) {
+                return new AutopopulateTemplate(template.getAbsolutePath());
+            }
+        }
+
+        if (workspace != null) {
```

Comment:
You already have a resource, which is placed in a workspace. This seems to allow loading from a workspace different than the resource one... what's the use case for this?

Reply:
The original intent was to look for a template valid at workspace level. I modev the check to look inside the resource workspace.